pkgs/development/compilers/vlang/default.nix

```diff
@@ -76,12 +96,32 @@ stdenv.mkDerivation rec {
     cp $HOME/vtest.v $out/lib/cmd/tools/vtest.v
   '';
 
-  meta = with lib; {
-    homepage = "https://vlang.io/";
-    description = "Simple, fast, safe, compiled language for developing maintainable software";
-    license = licenses.mit;
-    maintainers = with maintainers; [ Madouura ];
-    mainProgram = "v";
-    platforms = platforms.all;
-  };
+  preFixup = ''
+    # We need to patch the shebang of file "lib/cmd/tools/git_pre_commit_hook.vsh"
+    # and file "lib/vlib/v/tests/script_with_no_extension"
+    # Going from "#!/usr/bin/env -S v -raw-vsh-tmp-prefix tmp"
+    # To "#!$out/v -raw-vsh-tmp-prefix tmp"
+    sed -i "1s|.*|#!$out/bin/v -raw-vsh-tmp-prefix tmp|" $out/lib/cmd/tools/git_pre_commit_hook.vsh
+    sed -i "1s|.*|#!$out/bin/v -raw-vsh-tmp-prefix tmp|" $out/lib/vlib/v/tests/script_with_no_extension
+  '';
+
+  /* Details concerning features:
+    If you want to use `-compress`, you need to add `upx` in your build inputs.
+    If you want to build x11 apps (like examples/2048.v) you will need:
+    - `xorg.libX11.dev`
+    - `xorg.libXi.dev`
+    - `xorg.libXcursor.dev`
+    - `libGL.dev`
+    - `freetype`
```

Comment:
This should be toggable via a boolean instead.

Reply:
I should do so by adding a `buildWithExtras` as an input of the function, to make it available through `override` ?

Reply:
Fixed in https://github.com/COLAMAroro/nixpkgs/blob/dcaed354bd52104677eb30460088e627ce74e465/pkgs/development/compilers/vlang/default.nix#L54